playkit/src/main/java/com/kaltura/playkit/backend/phoenix/data/OttResultAdapter.java

```diff
@@ -32,9 +35,32 @@ public BaseResult deserialize(JsonElement json, Type typeOfT, JsonDeserializatio
         BaseResult baseResult = null;
 
         if(result != null && result.has("error")){
-             baseResult = new BaseResult(new Gson().fromJson(result.get("error"), ErrorElement.class));
+
+            ErrorElement error = new Gson().fromJson(result.get("error"), ErrorElement.class);
+
+            // trying to find constructor that excepts ErrorElement in order to return an object of type "typeOfT" even on error.
+            try {
+                Constructor<? extends Type> constructor = typeOfT.getClass().getConstructor(ErrorElement.class);
+
+                if (constructor != null) {
+                    baseResult = (BaseResult) constructor.newInstance(error);
+                }
+            } catch (NoSuchMethodException e) {
```

Comment:
i'm trying to create an object of type typeOfT with ErrorElement receiving constructor.
but not all classes ha this constructor so the base class will return.
that why i have empty catches, if the constructor not exists - do nothing